doc/man3/SSL_get_state.pod

```diff
@@ -0,0 +1,53 @@
+=pod
+
+=head1 NAME
+
+SSL_get_state, SSL_in_before, SSL_in_init, SSL_is_init_finished - Query the SSL handshake state
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ OSSL_HANDSHAKE_STATE SSL_get_state(const SSL *ssl);
+
+ int SSL_in_before(SSL *s);
+
+ int SSL_in_init(SSL *s);
+
+ int SSL_is_init_finished(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_get_state() gets the current B<ssl> handshake state.
+
+SSL_in_before() checks if an B<ssl> handshake has started.
+
+SSL_in_init() checks if an B<ssl> handshake is in progress.
```

Comment:
If I understand correctly, `SSL_in_before()` is kind of like `SSL_in_init()` but indicates that no wire communication has been made yet.